spec/models/allocation_spec.rb

```diff
@@ -32,16 +32,59 @@
     end
   end
 
-  describe '#refresh_moves_count!' do
-    let(:cancelled_move) { create :move, :cancelled }
-    let(:proposed_move) { create :move, :proposed }
-    let(:requested_move) { create :move, :requested }
-    let(:completed_move) { create :move, :completed }
-    let(:moves) { [cancelled_move, proposed_move, requested_move, completed_move] }
-    let!(:allocation) { create :allocation, moves: moves, moves_count: 1 }
+  describe '#refresh_status_and_moves_count!' do
+    context 'when updating moves count' do
+      let(:cancelled_move) { create :move, :cancelled }
+      let(:proposed_move) { create :move, :proposed }
+      let(:requested_move) { create :move, :requested }
+      let(:completed_move) { create :move, :completed }
+      let(:moves) { [cancelled_move, proposed_move, requested_move, completed_move] }
+      let!(:allocation) { create :allocation, moves: moves, moves_count: 1 }
+
+      it 'updates the number of non cancelled moves' do
+        expect { allocation.refresh_status_and_moves_count! }.to change { allocation.reload.moves_count }.from(1).to(3)
+      end
+    end
+
+    context 'when updating status' do
+      it 'sets status to `unfilled` if no profiles associated with uncancelled moves' do
+        moves = create_list(:move, 2, profile: nil)
+        cancelled_move = create(:move, :cancelled)
+        allocation = create(:allocation, moves: moves + [cancelled_move])
+
+        allocation.refresh_status_and_moves_count!
+        expect(allocation.reload).to be_unfilled
```

Comment:
don't `reload` - the allocation object should be in the right state after calling `refresh_status_and_moves_count!`